Cloudflare/zerotrust_dns/ingest/parser.yml

```
@@ -0,0 +1,154 @@
+name: zerotrust-dns
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: "{{original.message}}"
+  - name: set_ecs_fields
+  - name: set_cloudflare_fields
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          "@timestamp": '{{json_event.message.Datetime}}'
+          destination.ip: '{{json_event.message.DstIP}}'
+          destination.port: '{{json_event.message.DstPort}}'
+          device.id: '{{json_event.message.DeviceID}}'
+          dns.question.name: '{{json_event.message.QueryName}}'
+          dns.question.type: '{{json_event.message.QueryTypeName}}'
+          event.category: ["network"]
+          event.kind: 'event'
+          event.type: ["info"]
+          user.email: '{{json_event.message.Email}}'
+          user.id: '{{json_event.message.UserID}}'
+          source.ip: '{{json_event.message.SrcIP}}'
+          source.port: '{{json_event.message.SrcPort}}'
+          network.protocol: 'dns'
+          network.transport: '{{json_event.message.Protocol}}'
```

Add some constant and easy fields:
```suggestion
          network.transport: '{{json_event.message.Protocol}}'
          event.dataset: "gateway_dns"
          observer.vendor: "Cloudflare"
          observer.type: "dns"
```